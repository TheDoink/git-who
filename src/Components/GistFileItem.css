.fileItem {
	
	margin-top:10px;
	border-style: solid;
	border-width: 2px;
	border-color: black;
	background:silver;
}

.fileNameHeader {
	background:black;
}

.fileContent {
	margin:5px;
	overflow-x: auto;
	font-family: 'Source Code Pro', monospace;
	font-size:.75em;
	
	color:black;
}

.rowNum {
	padding-right:5px;
}

.contentRow {
	width:100%;
	white-space: pre;
}

.rowContent {
	padding-left:35px;
	
}