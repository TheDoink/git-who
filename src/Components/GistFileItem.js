import React, { Component } from 'react';
import './GistFileItem.css';

class GistFileItem extends Component {

  padNum(num) {
    let ret = num;
    while(ret.toString().length < this.props.numLines) {
      ret = ' '+ret;
    }
    return ret;
  };

  getContent() {
    let ret = [];
    if(this.props.file && this.props.file.content) {
      ret = this.props.file.content;
    }
    return ret;
  }

  render() {
    
    return (
        <div className="fileItem">
          <div className="fileNameHeader">
            {this.props.file.filename}
          </div>
          <div className="fileContent">
            {this.getContent.bind(this)().map((item, index) => (
              <div key={index} className="contentRow">
                <span className="rowNum">{this.padNum(index)}</span>
                <span className="rowContent">{item}</span>
              </div>
              
            ))}

            
          </div>
        </div>
    );
  }
}

export default GistFileItem;
