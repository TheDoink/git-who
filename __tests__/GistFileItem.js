// var NameLink = require('../src/Components/NameLink');
import React from 'react';
import renderer from 'react-test-renderer';
import GistFileItem from '../src/Components/GistFileItem';

const mockFile = {
      "filename": "Anagram.java",
      "type": "text/plain",
      "language": "Java",
      "raw_url": "https://gist.githubusercontent.com/jitun2004/015951870e6e47ecbd2372ce6afbc7b6/raw/6f7f94444471c2f50da79938efb830188e845f5e/Anagram.java",
      "size": 1095,
      "truncated": false,
      "content": ['package com.array;','','another line']
    };

it('lines up with the snapshot', () => {
  let testMe = renderer.create(<GistFileItem key="aUniqueFileName.txt" file={mockFile} numLines="3"></GistFileItem>).toJSON();

  expect(testMe).toMatchSnapshot();
  // testMe.getGist('d2eace7977e2aeaf942de0f8b3e7275c');

  
});
