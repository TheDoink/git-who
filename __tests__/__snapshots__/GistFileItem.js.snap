// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lines up with the snapshot 1`] = `
<div
  className="fileItem"
>
  <div
    className="fileNameHeader"
  >
    Anagram.java
  </div>
  <div
    className="fileContent"
  >
    <div
      className="contentRow"
    >
      <span
        className="rowNum"
      >
          0
      </span>
      <span
        className="rowContent"
      >
        package com.array;
      </span>
    </div>
    <div
      className="contentRow"
    >
      <span
        className="rowNum"
      >
          1
      </span>
      <span
        className="rowContent"
      >
        
      </span>
    </div>
    <div
      className="contentRow"
    >
      <span
        className="rowNum"
      >
          2
      </span>
      <span
        className="rowContent"
      >
        another line
      </span>
    </div>
  </div>
</div>
`;
